fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexFunctionHelper.java

```diff
@@ -110,6 +112,25 @@ private IndexFunctionHelper() {
                 .min(Comparator.comparing(i -> i.state.index.getColumnSize()));
     }
 
+    public static Optional<Pair<IndexAggregateFunction, IndexMaintainer>> enumerateIndexMaintainersForAggregateFunctionCall(@Nonnull FDBRecordStore store,
+                                                                                                                            @Nonnull IndexAggregateFunctionCall functionCall,
+                                                                                                                            @Nonnull List<String> recordTypeNames) {
+        return indexesForRecordTypes(store, recordTypeNames)
+                .filter(store::isIndexReadable)
+                .flatMap(index ->
+                        functionCall.enumerateIndexAggregateFunctionCandidates(index.getName())
+                                .flatMap(indexAggregateFunction -> {
+                                    final IndexMaintainer indexMaintainer = store.getIndexMaintainer(index);
+                                    return indexMaintainer.canEvaluateAggregateFunction(indexAggregateFunction)
+                                           ? Stream.of(Pair.of(indexAggregateFunction, store.getIndexMaintainer(index)))
```

Comment:
Isn't `store.getIndexMaintainer(index)` the same as `indexMaintainer`?

Reply:
Yup. Fixing.